code/datums/mind.dm

```diff
@@ -1527,6 +1530,52 @@
 	if(G)
 		G.reenter_corpse()
 
+/datum/mind/proc/make_zealot(mob/living/carbon/human/missionary)
+	if(!missionary || !istype(missionary))		//better provide a proper missionary or the rest of this is gonna break
+		return 0
+
+	zealot_master = missionary
+
+	var/list/implanters
+	var/ref = "\ref[missionary.mind]"
+	if(!(missionary.mind in ticker.mode:implanter))
+		ticker.mode:implanter[ref] = list()
+	implanters = ticker.mode:implanter[ref]
+	implanters.Add(src)
+	ticker.mode.implanted.Add(src)
+	ticker.mode.implanted[src] = missionary.mind
+	//ticker.mode:implanter[missionary.mind] += src
+	ticker.mode:implanter[ref] = implanters
+	ticker.mode.traitors += src
+	special_role = "traitor"
+	to_chat(current, "<span class='warning'><B>You're now a loyal zealot of [missionary.name]!</B> You now must lay down your life to protect them and assist in their goals at any cost.</span>")
+	var/datum/objective/protect/mindslave/MS = new
+	MS.owner = src
+	MS.target = missionary:mind

```

Comment:
see coder chat (summary: copied from mindslave implants with only variable name changes, code you last touched specifically)